File: flask/app/services/sentiment_analysis_service.py
class SentimentAnalysisService:
    POSITIVE_WORDS = ['bullish', 'moon', 'pump', 'partnership', 'growth', 'exciting']
    NEGATIVE_WORDS = ['bearish', 'dump', 'crash', 'regulation', 'delay', 'risk']

    @classmethod
    def analyze_sentiment(cls, text):
        text_lower = text.lower()
        
        positive_count = sum(1 for word in cls.POSITIVE_WORDS if word in text_lower)
        negative_count = sum(1 for word in cls.NEGATIVE_WORDS if word in text_lower)
        
        # Simple sentiment score calculation
        sentiment_score = positive_count - negative_count
        
        return {
            'score': sentiment_score,
            'positive_count': positive_count,
            'negative_count': negative_count
        }

File: server/src/services/SentimentAnalysisService.js
class SentimentAnalysisService {
    static analyzeSentiment(text) {
        // Simple sentiment analysis 
        const positiveWords = ['bullish', 'moon', 'pump', 'partnership', 'growth'];
        const negativeWords = ['bearish', 'dump', 'crash', 'regulation', 'delay'];

        const lowerText = text.toLowerCase();

        const positiveScore = positiveWords.filter(word => lowerText.includes(word)).length;
        const negativeScore = negativeWords.filter(word => lowerText.includes(word)).length;

        return {
            score: positiveScore - negativeScore,
            positiveCount: positiveScore,
            negativeCount: negativeScore
        };
    }
}